### ISSUE TITLE
Minus not working with Integer in qute after upgrading to 3.0.1.Final

### ISSUE THREAD

**Issue by Bazalii** [quarkusio/quarkus#32959]
### Describe the bug

After upgrading quarkus version to 3.0.1.Final, without changing anything else, appeared an error in qute templates:

```
io.quarkus.qute.TemplateException: Rendering error in template [HtmlPagesController/commentaries.html] line 60:
Method "minus(2)" not found on the base object "java.lang.Integer" in expression {commentariesRenderingInformation.renderingPageNumber.minus(2)}
```

It appeared in this line:
`{#let indexOfPreviousPage = commentariesRenderingInformation.renderingPageNumber.minus(2)}`

### Expected behavior

This expression worked fine until quarkus version upgrade, so the expected behavior is a correct subtraction.

### Actual behavior

Subtraction falls with an error

### Output of `java -version`

17.0.2

### Quarkus version or git rev

3.0.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.1

**Comment by mkouba** [comment]
> After upgrading quarkus version to 3.0.1.Final,...

What was the previous quarkus version?

**Comment by mkouba** [comment]
Hm, that's not a build time validation error but a runtime problem. That's weird as we have tests for this. Would you care to create a simple reproducer app?

**Comment by rasmushaglund** [comment]
I get the same kind of error for the `format` method, but only after a few live reloads:
```io.quarkus.qute.TemplateException: Rendering error in template [modules/Footer.qute.html] line 38: Method "format('Y')" not found on the base object "java.time.LocalDateTime" in expression {footer.date.format('Y')}```

Initially this works a few live reloads until it suddenly stops working and won't work again until I restart the Quarkus app. It always works the first time I load the site after starting Quarkus.

**Comment by mkouba** [comment]
> start the application

How do you start the app?

When I do `./gradlew build` in the root dir I get `java.util.NoSuchElementException: SRCFG00011: Could not expand value platform.quarkus.native.builder-image in property quarkus.native.builder-image`. When I try to build modules separately then I get `UnsatisfiedResolutionException: Unsatisfied dependency for type programmingLanguages.repositories.IProgrammingLanguageRepository` for `./gradlew core:build`. 

When I run the `web` module in the dev mode I can see the login page but commenting `@RolesAllowed("User", "HR", "Admin")` does not seem to have any effect, i.e. I can't get to the `/pages/resumes`.

By the way, I have noticed that your're using RESTEasy classic. You should switch to RESTEasy Reactive which is the preferred JAX-RS implementation for a long time (https://quarkus.io/blog/resteasy-reactive/).

**Comment by Bazalii** [comment]
>How do you start the app?

Sorry for not mentioning the command to start. I use `./gradlew web:build` or `./gradlew web:quarkusDev`. 

> I can't get to the /pages/resumes

Yes, it is not possible to open `/pages/resumes` because it is a page with pagination and requires query params, please try `/pages/resumes?pageIndex=0&pageSize=5`

>By the way, I have noticed that your're using RESTEasy classic. You should switch to RESTEasy Reactive which is the preferred 
JAX-RS implementation for a long time

I use classic resteasy because I faced another issue with qute-reactive. 

The idea behind resteasy-reactive is that if the signature of the method is synchronous (without using Uni, Multi, etc.) this method will be called in worker thread as expected for a synchronous method. I have only synchronous methods in my project that worked fine with resteasy-reactive. But when I added usage of qute-reactive I faced this error without making any of the methods async:
```
io.quarkus.runtime.BlockingOperationNotAllowedException: You have attempted to perform a blocking operation on a IO thread. 
This is not allowed, as blocking the IO thread will cause major performance issues with your application. 
If you want to perform blocking EntityManager operations make sure you are doing it from a worker thread.
        at io.quarkus.hibernate.orm.runtime.session.TransactionScopedSession.checkBlocking(TransactionScopedSession.java:116)
        at io.quarkus.hibernate.orm.runtime.session.TransactionScopedSession.createQuery(TransactionScopedSession.java:364)
        at org.hibernate.engine.spi.SessionLazyDelegator.createQuery(SessionLazyDelegator.java:548)
        at org.hibernate.engine.spi.SessionLazyDelegator.createQuery(SessionLazyDelegator.java:66)
        at org.hibernate.Session_3a974b6a18ac399f675913d732c105426414d370_Synthetic_ClientProxy.createQuery(Unknown Source)
        at io.quarkus.hibernate.orm.panache.common.runtime.CommonPanacheQueryImpl.createBaseQuery(CommonPanacheQueryImpl.java:353)
        at io.quarkus.hibernate.orm.panache.common.runtime.CommonPanacheQueryImpl.createQuery(CommonPanacheQueryImpl.java:332)
        at io.quarkus.hibernate.orm.panache.common.runtime.CommonPanacheQueryImpl.firstResult(CommonPanacheQueryImpl.java:281)
        at io.quarkus.hibernate.orm.panache.kotlin.runtime.PanacheQueryImpl.firstResult(PanacheQueryImpl.kt:122)
        at resumes.repositories.PanacheResumeRepository.getByUserId(PanacheResumeRepository.kt:23)
        at resumes.repositories.PanacheResumeRepository_ClientProxy.getByUserId(Unknown Source)
        at resumes.repositories.ResumeRepository.getByUserId(ResumeRepository.kt:45)
        at resumes.repositories.ResumeRepository_ClientProxy.getByUserId(Unknown Source)
        at resumes.services.implementations.ResumeService.getByUserId(ResumeService.kt:48)
        at resumes.services.implementations.ResumeService_ClientProxy.getByUserId(Unknown Source)
        at pages.controllers.HtmlPagesController.getMyResumePage(HtmlPagesController.kt:88)
        at pages.controllers.HtmlPagesController_ClientProxy.getMyResumePage(Unknown Source)
at pages.controllers.HtmlPagesController$quarkusrestinvoker$getMyResumePage_efed68af59cd5cc4e0387a005fb1da588c7aaea4.invoke(Unknown Source)
        at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
        at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)      
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:145)
        at org.jboss.resteasy.reactive.server.handlers.RestInitialHandler.beginProcessing(RestInitialHandler.java:48)
        at org.jboss.resteasy.reactive.server.vertx.ResteasyReactiveVertxHandler.handle(ResteasyReactiveVertxHandler.java:23)
        at org.jboss.resteasy.reactive.server.vertx.ResteasyReactiveVertxHandler.handle(ResteasyReactiveVertxHandler.java:10)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
        at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
        at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:102)
        at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:88)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:140)
        at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
        at io.vertx.ext.web.handler.impl.StaticHandlerImpl.lambda$sendStatic$1(StaticHandlerImpl.java:290)
        at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
        at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

For example, `getCommentariesPage` method of `HtmlPagesController`, which handles a request and includes synchronous database call, causes this error. You can get this behavior using the same reproducer. If you want to, please change `quarkus-resteasy` , `quarkus-resteasy-jsonb` in core module and `quarkus-resteasy-qute` in web module to reactive versions and go to the `/pages/resumes?pageIndex=0&pageSize=5`

**Comment by mkouba** [comment]
> Yes, it is not possible to open `/pages/resumes` because it is a page with pagination and requires query params, please try `/pages/resumes?pageIndex=0&pageSize=5`

Hm, query params don't help...

> But when I added usage of qute-reactive I faced this error without making any of the methods async...

Yes, the `TemplateInstance` used as a return type implies a non-blocking resource method. You can add `@Blocking` to the resource method or resource class to change the default behavior.

**Comment by Bazalii** [comment]
> > Yes, it is not possible to open `/pages/resumes` because it is a page with pagination and requires query params, please try `/pages/resumes?pageIndex=0&pageSize=5`
> 
> Hm, query params don't help...
> 
> > But when I added usage of qute-reactive I faced this error without making any of the methods async...
> 
> Yes, the `TemplateInstance` used as a return type implies a non-blocking resource method. You can add `@Blocking` to the resource method or resource class to change the default behavior.

Oh, sorry again. Try `/pages/commentaries?pageIndex=0&pageSize=5`. 152 line closes access to `commentaries` page

**Comment by mkouba** [comment]
> Oh, sorry again. Try `/pages/commentaries?pageIndex=0&pageSize=5`. 152 line closes access to `commentaries` page

No problem. I managed to comment the line 119 ;-). But still no success... Isn't there any other security config, e.g. in `application.properties`?

**Comment by Bazalii** [comment]
> > Oh, sorry again. Try `/pages/commentaries?pageIndex=0&pageSize=5`. 152 line closes access to `commentaries` page
> 
> No problem. I managed to comment the line 119 ;-). But still no success... Isn't there any other security config, e.g. in `application.properties`?

I forgot that I also have a redirection interceptor, but there is one more problem. There are some fields that need to be injected in the controller after jwt token parse, so, maybe you will just "register" in the application(there is no email confirmation, so no real information is needed)? Or I may create another reproducer

**Comment by mkouba** [comment]
> Or I may create another reproducer

Please do ;-)

**Comment by Bazalii** [comment]
> > Or I may create another reproducer
> 
> Please do ;-)

I have added reproducer to logn.html file. Please, get the latest commit of the same branch, start the application as before and go to `/pages/login`(it will wotk fine), after that trigger live reload(for example, comment line 152 of HtmlPagesController) and refresh the page and you will get the error

**Comment by mkouba** [comment]
Thanks. I was able to reproduce the problem with [qute-quickstart](https://github.com/quarkusio/quarkus-quickstarts/tree/main/qute-quickstart). It's actually a regression introduced in https://github.com/quarkusio/quarkus/pull/32653 (3.0.1.Final). I'm working on a fix.

**Comment by Bazalii** [comment]
> Thanks. I was able to reproduce the problem with [qute-quickstart](https://github.com/quarkusio/quarkus-quickstarts/tree/main/qute-quickstart). It's actually a regression introduced in #32653 (3.0.1.Final). I'm working on a fix.

Thank you. I laso have one more question which is related to the topic of this issue a little bit. Is it an expected behavior that it is not possible to use minus(-) for assigning values to qute variables like this:
{#let indexOfPreviousPage = commentariesRenderingInformation.renderingPageNumber - 2}
Because in this case I get this exception:
`HtmlPagesController/commentaries.html:60:14 - {-}: Only type-safe expressions are allowed in the checked template defined via: pages.controllers.Htm
lPagesController$Templates.commentaries(); an expression must be based on a checked template parameter [commentariesRenderingInformation], or bound via a param 
declaration, or the requirement must be relaxed via @CheckedTemplate(requireTypeSafeExpressions = false)`
And I have to change this statement to this one:
{#let indexOfPreviousPage = commentariesRenderingInformation.renderingPageNumber.minus(2)}
or to this one: 
{#let indexOfPreviousPage = (commentariesRenderingInformation.renderingPageNumber - 2)}
Am I missing something?

**Comment by mkouba** [comment]
> Is it an expected behavior that it is not possible to use minus(-) for assigning values to qute variables like this:
{#let indexOfPreviousPage = commentariesRenderingInformation.renderingPageNumber - 2}

You can only the minus `-` sign in output expressions; i.e. `{commentariesRenderingInformation.renderingPageNumber - 2}` is supported. In section params you either need to use the method notation (`commentariesRenderingInformation.renderingPageNumber.minu(2)`) or parentheses: `{#let indexOfPreviousPage = (commentariesRenderingInformation.renderingPageNumber - 2)}` (see https://quarkus.io/guides/qute-reference#sections).

**Comment by Bazalii** [comment]
> Yes, the `TemplateInstance` used as a return type implies a non-blocking resource method. You can add `@Blocking` to the resource method or resource class to change the default behavior.

@mkouba, are you sure that Blocking annotation solves the problem? I have added Blocking annotation but I still get the same error
```
    @APIResponses(
        APIResponse(responseCode = "200", description = "Successful operation")
    )
    @GET
    @Path("resumes")
    @Produces(MediaType.TEXT_HTML)
    @RolesAllowed("User", "HR", "Admin")
    @Blocking
    fun getResumesPage(
        @QueryParam("pageIndex") pageIndex: Int,
        @QueryParam("pageSize") pageSize: Int,
    ): TemplateInstance {
        val resumes = _resumeService.getPage(pageIndex, pageSize)
        val resumesNumber = _resumeService.countAll()
        var numberOfPages = (resumesNumber / pageSize).toInt()

        if ((resumesNumber % pageSize).toInt() != 0) {
            numberOfPages += 1
        }

        return Templates.resumes(
            ResumesRenderingInformation(
                resumes.map {
                    ResumeRenderingInformation(it, _userService.getById(it.userId).email)
                },
                pageIndex + 1,
                numberOfPages,
                _groups,
                _userId,
                _requestStartTime
            )
        )
    }
```

```
io.quarkus.runtime.BlockingOperationNotAllowedException: You have attempted to perform a blocking operation on a IO thread. This is not allowed, as blocking the IO thread will cause major performance issues with your application. If you want to perform blocking EntityM
anager operations make sure you are doing it from a worker thread.
        at io.quarkus.hibernate.orm.runtime.session.TransactionScopedSession.checkBlocking(TransactionScopedSession.java:116)
        at io.quarkus.hibernate.orm.runtime.session.TransactionScopedSession.createQuery(TransactionScopedSession.java:364)
        at org.hibernate.engine.spi.SessionLazyDelegator.createQuery(SessionLazyDelegator.java:548)
        at org.hibernate.engine.spi.SessionLazyDelegator.createQuery(SessionLazyDelegator.java:66)
        at org.hibernate.Session_3a974b6a18ac399f675913d732c105426414d370_Synthetic_ClientProxy.createQuery(Unknown Source)
        at io.quarkus.hibernate.orm.panache.common.runtime.CommonPanacheQueryImpl.createBaseQuery(CommonPanacheQueryImpl.java:353)
        at io.quarkus.hibernate.orm.panache.common.runtime.CommonPanacheQueryImpl.createQuery(CommonPanacheQueryImpl.java:315)
        at io.quarkus.hibernate.orm.panache.common.runtime.CommonPanacheQueryImpl.list(CommonPanacheQueryImpl.java:266)
        at io.quarkus.hibernate.orm.panache.kotlin.runtime.PanacheQueryImpl.list(PanacheQueryImpl.kt:114)
        at resumes.repositories.PanacheResumeRepository.getPage(PanacheResumeRepository.kt:27)
        at resumes.repositories.PanacheResumeRepository_ClientProxy.getPage(Unknown Source)
        at resumes.repositories.ResumeRepository.getPage(ResumeRepository.kt:51)
        at resumes.repositories.ResumeRepository_ClientProxy.getPage(Unknown Source)
        at resumes.services.implementations.ResumeService.getPage(ResumeService.kt:52)
        at resumes.services.implementations.ResumeService_ClientProxy.getPage(Unknown Source)
        at pages.controllers.HtmlPagesController.getResumesPage(HtmlPagesController.kt:127)
        at pages.controllers.HtmlPagesController_Subclass.getResumesPage$$superforward(Unknown Source)
        at pages.controllers.HtmlPagesController_Subclass$$function$$4.apply(Unknown Source)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:74)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext$NextAroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:100)
        at io.quarkus.security.runtime.interceptor.SecurityHandler.handle(SecurityHandler.java:27)
        at io.quarkus.security.runtime.interceptor.RolesAllowedInterceptor.intercept(RolesAllowedInterceptor.java:29)
        at io.quarkus.security.runtime.interceptor.RolesAllowedInterceptor_Bean.intercept(Unknown Source)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:71)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:63)
        at io.quarkus.resteasy.reactive.server.runtime.StandardSecurityCheckInterceptor.intercept(StandardSecurityCheckInterceptor.java:44)
        at io.quarkus.resteasy.reactive.server.runtime.StandardSecurityCheckInterceptor_RolesAllowedInterceptor_Bean.intercept(Unknown Source)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:42)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:38)
        at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:26)
        at pages.controllers.HtmlPagesController_Subclass.getResumesPage(Unknown Source)
        at pages.controllers.HtmlPagesController_ClientProxy.getResumesPage(Unknown Source)
        at pages.controllers.HtmlPagesController$quarkusrestinvoker$getResumesPage_4271c82a5b54e47cca175b709859d7f8c14d3233.invoke(Unknown Source)
        at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
        at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)      
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:145)
        at org.jboss.resteasy.reactive.server.handlers.RestInitialHandler.beginProcessing(RestInitialHandler.java:48)
        at org.jboss.resteasy.reactive.server.vertx.ResteasyReactiveVertxHandler.handle(ResteasyReactiveVertxHandler.java:23)
        at org.jboss.resteasy.reactive.server.vertx.ResteasyReactiveVertxHandler.handle(ResteasyReactiveVertxHandler.java:10)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:177)
        at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
        at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:102)
        at io.quarkus.vertx.http.runtime.StaticResourcesRecorder$2.handle(StaticResourcesRecorder.java:88)
        at io.vertx.ext.web.impl.RouteState.handleContext(RouteState.java:1284)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:140)
        at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:141)
        at io.vertx.ext.web.handler.impl.StaticHandlerImpl.lambda$sendStatic$1(StaticHandlerImpl.java:290)
        at io.vertx.core.impl.future.FutureImpl$3.onSuccess(FutureImpl.java:141)
        at io.vertx.core.impl.future.FutureBase.lambda$emitSuccess$0(FutureBase.java:54)
        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)

```

I have alsoi tried adding Blocking annotation to controller and behavior is the same..
